src/node/rpc/node_frontend.h

```diff
@@ -289,6 +289,35 @@ namespace ccf
             return make_success(rep);
           }
 
+          if (
+            consensus != nullptr && consensus->type() == ConsensusType::CFT &&
+            !this->context.get_node_state().is_primary())
+          {
+            auto primary_id = consensus->primary();
+            if (primary_id.has_value())
+            {
+              auto nodes = args.tx.ro(this->network.nodes);
+              auto info = nodes->get(primary_id.value());
+              if (info)
+              {
+                args.rpc_ctx->set_response_header(
+                  "Location",
+                  fmt::format(
+                    "https://{}:{}/node/join", info->pubhost, info->pubport));
+
+                return make_error(
+                  HTTP_STATUS_PERMANENT_REDIRECT,
+                  ccf::errors::NodeRetired,
```

Comment:
This isn't always true, is it? We could get here if the joiner's target node is not/no longer primary, but isn't necessarily retired.

Reply:
Yes!